### PR TITLE
Updated Edit Profile page to use newer `<CWForm/>` component with newer validations + cleaned up related stale components

### PR DESCRIPTION
**Pull request by mzparacha**
## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/7935

## Description of Changes
- Updated Edit Profile page to use newer `<CWForm/>` component with newer validations
- Cleaned up related stale components
- Added CWForm/form-validation compatibility to `ReactQuillEditor`
- DRY'ed related code from other areas of the app.

## "How We Fixed It"
N/A

## Test Plan

1. Visit the edit profile page
2. CA - Verify you can perform all the edit actions as before and that there are no unexpected issues
3. Imp: during step 2, rigorously test the quill editor and its validation states to make sure there are no inconsistencies or issues as this is one of the main components.

## Deployment Plan
N/A

## Other Considerations
There is currently a bug in production that prevents users from updating their profile avatar, this will be addressed in https://github.com/hicommonwealth/commonwealth/issues/8151

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/components/EditProfile/EditProfile.tsx [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+                    }}
+                  />
+                </div>
+              </div>
+              <CWTextInput
+                fullWidth
+                placeholder="Enter your user name"
+                label="Username"
+                // TODO: unique username like in PersonalInformationStep?
+                // TODO: username generator like in PersonalInformationStep?
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/EditProfile/EditProfile.tsx:328]: Will have to confirm if we want to add this logic here. This is just for enhancement and not a blocker.

In case we want to add this logic, it will be in a separate PR.

File: packages/commonwealth/client/scripts/views/components/EditProfile/EditProfile.tsx [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+                onLinkRemovedAtIndex={onLinkRemovedAtIndex}
+                canAddLinks={links.length <= 5}
+              />
+            </ProfileSection>
+            <ProfileSection
+              title="Personalize Your Profile"
+              description="Express yourself through imagery."
+            >
+              <CWText fontWeight="medium">Add a background image </CWText>
+              {/* TODO: add option to remove existing image */}
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/EditProfile/EditProfile.tsx:373]: Another potential enhancement, this behavior is not yet implemented.

I feel we would want this, will be creating a new ticket for this.

**Review by masvelio — COMMENTED** [review]
Getting bug on profile page load
image

image


Cant see anymore option to select tile vs full image

image

I can set my name as Anonymous, is this correct?

image


Besides, AMAZING WORK, thanks for all refactors and proactive attitude!

File: packages/commonwealth/client/scripts/views/components/EditProfile/EditProfile.tsx [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+  url: string;
+  imageBehavior: ImageBehavior;
+};
+
+const EditProfile = () => {
+  const userOnboardingEnabled = useFlag('userOnboardingEnabled');
+  const navigate = useNavigate();
+  const { isLoggedIn } = useUserLoggedIn();
+  const [profile, setProfile] = useState<NewProfile>();
+  const [avatarUrl, setAvatarUrl] = useState();
```
[the remark this answers is not in the document]
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/components/EditProfile/EditProfile.tsx:53]: this was actually the same in pre-cleanup code. Also the avatar setting logic is broken, which I plan to fix in https://github.com/hicommonwealth/commonwealth/issues/8151

**Comment by mzparacha** [comment]
> Getting bug on profile page load
 
good catch, fixed.

> Cant see anymore option to select tile vs full image

fixed in https://github.com/hicommonwealth/commonwealth/pull/8150/commits/a0f88652a612e7b3a7f2d7f66fb7576779dc9ca8

> I can set my name as Anonymous, is this correct?

For now, yes. I only migrated code to newer standards. Would love to add the username uniqueness / random generation, more context [here](https://github.com/hicommonwealth/commonwealth/pull/8150#discussion_r1639865537)

---

Thanks for the feedback. Fixed all the issues.

**Review by masvelio — APPROVED** [review]

**Review by Israellund — APPROVED** [review]
